src/core/cgroup.c

```diff
@@ -3510,6 +3510,34 @@ void manager_invalidate_startup_units(Manager *m) {
                 unit_invalidate_cgroup(u, CGROUP_MASK_CPU|CGROUP_MASK_IO|CGROUP_MASK_BLKIO);
 }
 
+int compare_job_cpu_weight(const void *a, const void *b) {
+        const Job *x = a, *y = b;
+        ManagerState state = manager_state(x->unit->manager);
+        CGroupContext *context;
+        uint64_t weight_x, weight_y;
+        int ret;
+
+        context = unit_get_cgroup_context(x->unit);
+        weight_x = context ? cgroup_context_cpu_weight(context, state) : CGROUP_WEIGHT_DEFAULT;
+        context = unit_get_cgroup_context(y->unit);
+        weight_y = context ? cgroup_context_cpu_weight(context, state) : CGROUP_WEIGHT_DEFAULT;
```

Comment:
please hide these two operations in a new helper call: 

```c
uint64_t unit_get_cpu_weight(Unit *u, …);
```

Which does the two steps internally

Reply:
ok